File: codility/lessons/4-counting_elements/max_counters/solution.c
struct Results solution(int number_of_counters, int operations[], int number_of_operations) {
    int* counters = (int*) calloc(number_of_counters, sizeof(int));

    int max_count = 0;

    for (int i = 0; i < number_of_operations; ++i) {
        if (operations[i] <= number_of_counters) {
            int counter_index = operations[i] - 1;

            if (max_count == counters[counter_index]) {
                ++max_count;
            }

            ++counters[counter_index];
        } else {
            for (int j = 0; j < number_of_counters; ++j) {
                counters[j] = max_count;
            }
        }
    }

    struct Results result;
    result.C = counters;
    result.L = number_of_counters;

    return result;
}

File: codility/lessons/4-counting_elements/max_counters/solution.js
function solution(numberOfCounters, operations) {
    var counters = Array.apply(null, Array(numberOfCounters)).map(Number.prototype.valueOf, 0);
    var maxCount = 0;
    var i, j;

    for (i = 0; i < operations.length; ++i) {
        if (operations[i] <= numberOfCounters) {
            if (maxCount == counters[operations[i] - 1]) {
                ++maxCount;
            }

            ++counters[operations[i] - 1];
        } else {
            for (j = 0; j < counters.length; ++j) {
                counters[j] = maxCount;
            }
        }
    }

    return counters;
}
